components/shared_code/tests/shared/model/test_measurement.py

```diff
@@ -250,6 +250,37 @@ def test_entity_user_data(self):
             if source["source_uuid"] == SOURCE_ID:
                 self.assertTrue("entity_user_data" in source)
 
+    def test_copy_first_seen_timestamps(self):
+        """Test that the first seen timestamps can be copied."""
+        measurement_1 = Measurement(
+            self.metric(),
+            sources=[
+                {
+                    "source_uuid": SOURCE_ID,
+                    "type": "azure_devops",
+                    "entities": [{"key": "key1", "first_seen": "2023-07-15"}],
+                },
+            ],
+        )
+        measurement_2 = Measurement(
+            self.metric(),
+            sources=[
+                {
+                    "source_uuid": SOURCE_ID,
+                    "type": "azure_devops",
+                    "entities": [{"key": "key1", "first_seen": "2023-07-16"}],
```

Comment:
Perhaps good to set this date to "2023-07-17", so it isn't the exact same as source 2